core/src/test/java/com/zextras/carbonio/files/message_broker/UserStatusChangedIT.java

```diff
@@ -0,0 +1,121 @@
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package com.zextras.carbonio.files.message_broker;
+
+import com.google.inject.Injector;
+import com.zextras.carbonio.files.Simulator;
+import com.zextras.carbonio.files.Simulator.SimulatorBuilder;
+import com.zextras.carbonio.files.api.utilities.DatabasePopulator;
+import com.zextras.carbonio.files.api.utilities.entities.SimplePopulatorTextFile;
+import com.zextras.carbonio.files.dal.dao.ebean.Node;
+import com.zextras.carbonio.files.dal.repositories.interfaces.NodeRepository;
+import com.zextras.carbonio.files.message_broker.consumers.UserStatusChangedConsumer;
+import com.zextras.carbonio.message_broker.events.services.mailbox.UserStatusChanged;
+import com.zextras.carbonio.message_broker.events.services.mailbox.enums.UserStatus;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Here I mock an event emitted and received because if I was to emit a real event I could not know
+ * when it will be consumed
+ */
+class UserStatusChangedIT {
+
+  static Simulator simulator;
+  static NodeRepository nodeRepository;
+
+  @BeforeAll
+  static void init() {
+    Map<String, String> users = new HashMap<String, String>();
+    users.put("fake-token", "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa");
+    users.put("fake-token2", "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaab");
+    users.put("fake-token3", "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaac");
+    simulator =
+        SimulatorBuilder.aSimulator()
+            .init()
+            .withDatabase()
+            .withMessageBroker()
+            .withServiceDiscover()
+            .withUserManagement(users)
+            .build()
+            .start();
+    final Injector injector = simulator.getInjector();
+    nodeRepository = injector.getInstance(NodeRepository.class);
+  }
+
+  @AfterEach
+  void cleanUp() {
+    simulator.resetDatabase();
+  }
+
+  @AfterAll
+  static void cleanUpAll() {
+    simulator.stopAll();
+  }
+
+  @Test
+  void
+  givenANotHiddenNodeAndAnUserStatusChangedEventClosedWrittenOnMessageBrokerQueueUsersNodesHiddenFlagsShouldBeTrue() {
+    // Given
+    DatabasePopulator.aNodePopulator(simulator.getInjector())
+        .addNode(
+            new SimplePopulatorTextFile(
+                "00000000-0000-0000-0000-000000000000", "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"));
+
+    // When
+    UserStatusChangedConsumer userStatusChangedConsumer = new UserStatusChangedConsumer(nodeRepository);
+    userStatusChangedConsumer.doHandle(new UserStatusChanged("aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa", UserStatus.CLOSED));
+
+    // Then
+    Optional<Node> nodeOpt = nodeRepository.getNode("00000000-0000-0000-0000-000000000000");
+    Assertions.assertThat(nodeOpt.isPresent() && nodeOpt.get().isHidden()).isTrue();
```

Comment:
You can put the `idNodeHidden` condition as a class member, initialized in the setup class and use it in all class tests. I think the assertion becomes more readable

```suggestion
    Condition<Node> isNodeHidden = new Condition<Node>(node -> node.isHidden(), "node is hidden");
    Assertions.assertThat(nodeOpt).isPresent().is(isNodeHidden);
```